app/src/main/java/com/github/geohunt/app/ui/screens/main/MainViewModel.kt

```diff
@@ -41,6 +50,32 @@ class MainViewModel(
         }
     }
 
+    fun listenForNewNotification(context: Context) {
+        val database = FirebaseDatabase.getInstance()
+        val challengesRef = database.getReference("challenges")
+
+        val challengesListener = object : ValueEventListener {
+            override fun onDataChange(snapshot: DataSnapshot) {
+                snapshot.children.forEach { child ->
+                    if (child.key != null) {
+                        showNotification(
+                            context = context,
+                            title = "New Challenge!",
```

Comment:
Maybe use `stringResource` for i18n?

Reply:
I'm not sure I understand what you mean. We don't need any formatting here, it's just passing title and message for notification. Do you mean a constants file of sort and define the title and message there and then use it here?

Reply:
Basically, every text should be put in the `strings.xml` file to make internationalization a thing.